### PR TITLE
Allow users to check addresses and coins on chain

### PR DESCRIPTION
**Pull request by jack60612**
The change from print_transaction to print_coins_transactions was because some of the logic needs to be reused and mariano suggested to combine them into one function. 
These changes in total, allow a user to check any puzzle hash (address) or any coin id for the transaction history or other information, this really just adds some more features and convenience to the cli.

### CODE REVIEW

**Review by mariano54 — CHANGES_REQUESTED** [review]
- Rename to coins not tx
- Integrate into show
- support by address and by coin id
- execute with node should be extracted into a new function
- confirmed index and spent index
- parent coin ID

**Comment by jack60612** [comment]
all tested and working

File: chia/cmds/wallet_funcs.py [hunk 1]
```diff
@@ -496,7 +481,7 @@ async def get_wallet(wallet_client: WalletRpcClient, fingerprint: int = None) ->
     else:
         print("Choose wallet key:")
         for i, fp in enumerate(fingerprints):
-            print(f"{i+1}) {fp}")
+            print(f"{i + 1}) {fp}")
```
Comment (emlowe) [inline chia/cmds/wallet_funcs.py:484]: same?

File: chia/cmds/wallet_funcs.py [hunk 2]
```diff
@@ -450,7 +435,7 @@ def wallet_coin_unit(typ: WalletType, address_prefix: str) -> Tuple[str, int]:
 
 
 def print_balance(amount: int, scale: int, address_prefix: str) -> str:
-    ret = f"{amount/scale} {address_prefix} "
+    ret = f"{amount / scale} {address_prefix} "
```
Comment (emlowe) [inline chia/cmds/wallet_funcs.py:438]: same here, unrelated change?

File: chia/cmds/wallet_funcs.py [hunk 3]
```diff
@@ -97,8 +82,8 @@ async def get_transaction(args: dict, wallet_client: WalletRpcClient, fingerprin
         print(e.args[0])
         return
 
-    print_transaction(
-        tx,
+    print_coins_transactions(
```
Comment (emlowe) [inline chia/cmds/wallet_funcs.py:85]: I'd would have probalby not changed this function as it's orthogonal to the functionality you are adding.

Comment (jack60612) [reply chia/cmds/wallet_funcs.py:85]: oops i see what i did

Comment (jack60612) [reply chia/cmds/wallet_funcs.py:85]: ignore the comment that i deleted, that was unrelated

Comment (emlowe) [reply chia/cmds/wallet_funcs.py:85]: I guess I don't understand because I see print_coins_transactions added to show.py and it literally does `if (a) do a else do b` so afaict your duped it anyway

Comment (jack60612) [reply chia/cmds/wallet_funcs.py:85]: > I guess I don't understand because I see print_coins_transactions added to show.py and it literally does `if (a) do a else do b` so afaict your duped it anyway

no wait, fair point

Comment (jack60612) [reply chia/cmds/wallet_funcs.py]: fixed it, thank you for that.

File: chia/cmds/show.py [hunk 4]
```diff
@@ -1,13 +1,45 @@
...
-    connections = await client.get_connections()
+def print_coins(
+    verbose: bool,
+    name: str,
+    address_prefix: str,
+    mojo_per_unit: int,
+    coin_record: CoinRecord,
+) -> None:
+    if verbose:
+        print(coin_record)
```
Comment (mariano54) [inline chia/cmds/show.py:23]: probably better to print in JSON:
coin_record.to_json_dict()

File: chia/cmds/show.py [hunk 5]
```diff
@@ -1,13 +1,45 @@
...
+    else:
+        chia_amount = Decimal(int(coin_record.coin.amount)) / mojo_per_unit
+        coin_address = encode_puzzle_hash(coin_record.coin.puzzle_hash, address_prefix)
+        confirmed_block = coin_record.confirmed_block_index
+        print(f"Coin 0x{coin_record.name}")
+        print(f"Wallet Address: {coin_address}")
+        print(
+            "Status: "
+            f"{f'Confirmed at block: {confirmed_block}' if coin_record.confirmed_block_index != 0 else 'Pending'}"
```
Comment (mariano54) [inline chia/cmds/show.py:32]: Maybe "Not found" instead of Pending

File: chia/cmds/show.py [hunk 6]
```diff
@@ -1,13 +1,45 @@
...
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_record import CoinRecord
+from chia.util.bech32m import encode_puzzle_hash
+from chia.util.ints import uint64
 from chia.util.network import is_trusted_inner
 
 
-async def print_connections(client, time, NodeType, trusted_peers: Dict):
-    connections = await client.get_connections()
+def print_coins(
```
Comment (mariano54) [inline chia/cmds/show.py:15]: print_coin_record instead of print_coins?

File: chia/cmds/show.py (hunk 4 above)
Comment (mariano54) [inline chia/cmds/show.py:23]: probably better to print in JSON:
coin_record.to_json_dict()

File: chia/cmds/show.py [hunk 7]
```diff
@@ -1,13 +1,45 @@
...
+        confirmed_block = coin_record.confirmed_block_index
+        print(f"Coin 0x{coin_record.name}")
+        print(f"Wallet Address: {coin_address}")
+        print(
+            "Status: "
+            f"{f'Confirmed at block: {confirmed_block}' if coin_record.confirmed_block_index != 0 else 'Pending'}"
+        )
+        print(f"Spent: {f'At Block {coin_record}' if not coin_record.spent else 'No'}")
+        print(f"Amount {'sent'}: {chia_amount} {name}")
+        print(f"Parent Coin ID: 0x{coin_record.coin.parent_coin_info}")
```
Comment (mariano54) [inline chia/cmds/show.py:36]: `parent_coin_info.hex()`

File: chia/cmds/show.py [hunk 8]
```diff
@@ -1,13 +1,45 @@
...
+        chia_amount = Decimal(int(coin_record.coin.amount)) / mojo_per_unit
+        coin_address = encode_puzzle_hash(coin_record.coin.puzzle_hash, address_prefix)
+        confirmed_block = coin_record.confirmed_block_index
+        print(f"Coin 0x{coin_record.name}")
+        print(f"Wallet Address: {coin_address}")
+        print(
+            "Status: "
+            f"{f'Confirmed at block: {confirmed_block}' if coin_record.confirmed_block_index != 0 else 'Pending'}"
+        )
+        print(f"Spent: {f'At Block {coin_record}' if not coin_record.spent else 'No'}")
```
Comment (mariano54) [inline chia/cmds/show.py:34]: `at` not `At`

File: chia/cmds/show.py [hunk 9]
```diff
@@ -1,13 +1,45 @@
...
+    mojo_per_unit: int,
+    coin_record: CoinRecord,
+) -> None:
+    if verbose:
+        print(coin_record)
+    else:
+        chia_amount = Decimal(int(coin_record.coin.amount)) / mojo_per_unit
+        coin_address = encode_puzzle_hash(coin_record.coin.puzzle_hash, address_prefix)
+        confirmed_block = coin_record.confirmed_block_index
+        print(f"Coin 0x{coin_record.name}")
```
Comment (mariano54) [inline chia/cmds/show.py:28]: `name.hex()`

File: chia/cmds/show.py [hunk 10]
```diff
@@ -57,236 +89,305 @@ async def print_connections(client, time, NodeType, trusted_peers: Dict):
...
+
+    import aiohttp
+
+    from chia.util.config import load_config
+    from chia.util.default_root import DEFAULT_ROOT_PATH
+    from chia.util.ints import uint16
+    from chia.rpc.full_node_rpc_client import FullNodeRpcClient
+
+    try:
+        config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
```
Comment (mariano54) [inline chia/cmds/show.py:103]: we should use the config loading methods here now

File: chia/cmds/show.py [hunk 11]
```diff
@@ -57,236 +89,305 @@ async def print_connections(client, time, NodeType, trusted_peers: Dict):
...
+            print("This is normal if full node is still starting up")
+        else:
+            tb = traceback.format_exc()
+            print(f"Exception from 'show' {tb}")
+
+    node_client.close()
+    await node_client.await_closed()
+
+
 async def show_async(
```
Comment (mariano54) [inline chia/cmds/show.py:124]: This method is getting very long and doing various things. This should be decomposed into smaller methods, one for each command.

**Review by mariano54 — CHANGES_REQUESTED** [review]

File: chia/cmds/show.py (hunk 7 above)
Comment (jack60612) [reply chia/cmds/show.py:36]: i assumed that printing it would auto hex it but ill add that anyway

File: chia/cmds/show.py (hunk 10 above)
Comment (jack60612) [reply chia/cmds/show.py:103]: Im not sure what you mean by that.

**Comment by jack60612** [comment]
Wait why did i close this?

**Comment by jack60612** [comment]
for ci

**Comment by mariano54** [comment]
Actually, I realized we already have the following:
```
chia rpc full_node get_coin_record_by_name '{"name":"0x7e10a77d0d0ec1074fa5348dd03bdcd11a1e2dbd57b5ab6e89b888542f86ce20"}'
```

```
chia rpc full_node get_coin_records_by_puzzle_hash '{"puzzle_hash":"0x7e10a77d0d0ec1074fa5348dd03bdcd11a1e2dbd57b5ab6e89b888542f86ce20"}'
```

I think it would be redundant to add another way to fetch these
